src/assert.js

```diff
@@ -14,6 +14,14 @@ class Assert {
 
 	// Assert helpers
 
+	timeout( duration ) {
+		if ( objectType( duration ) !== "number" ) {
```

Comment:
I don't think we should support that. I'd rather have it throw an exception. These are not wild values, they are crafted for the specific purpose of setting a timeout in QUnit.